internal/handlers/common/aggregations/aggregations.go

```diff
@@ -24,6 +24,31 @@ import (
 
 // Stage is a common interface for all aggregation stages.
 type Stage interface {
+	// FetchDocuments fetches document iterator, it is only used if this is
+	// the first stage of the pipeline.
+	// The returned document iterator may be originated from querying database or
+	// from in-memory value iterator.
+	// This allows first stage of the pipeline to decide how to create initial document iterator.
+	FetchDocuments(ctx context.Context, closer *iterator.MultiCloser) (types.DocumentsIterator, error)
+
 	// Process applies an aggregate stage on documents from iterator.
 	Process(ctx context.Context, iter types.DocumentsIterator, closer *iterator.MultiCloser) (types.DocumentsIterator, error)
 }
+
+// AggregateQuery is a common interface for fetching from database.
+type AggregateQuery interface {
```

Comment:
So, this interface will grow when we have `listSessions` and other stages that need different types of documents. I think it's fine. This way, we have a unified interface to work with different backends. 

The approach looks similar to the `Backend`, `Collection`, and other interfaces that we have in the new handler.
Maybe we can have a similar naming too? And then this could be just `Aggregate`/`Aggregation`? But I think the naming it's not so critical.